docs/docs/vm-fundamentals/hints.md

```diff
@@ -2,6 +2,137 @@
 sidebar_position: 3
 ---
 
-# Hints
 
-There are several hints, explain each of them here. How they interact with the VM and affects it.
+# Hint
+  ## HINTS  IN CAIRO VIRTUAL MACHINE
+Hints in Cairo are pieces of Python code only executed by the sequencer.
+
+Hints are used to help the Virtual Machine determine the next path of execution it should take.
+
+Below are some Cairo code with hints:
+
+```cairo
+      [ap] = 25, ap++;
+    %{
+        import math
+         memory[ap] = int(math.sqrt(memory[ap - 1]))
+     %}
+      [ap - 1] = [ap] * [ap], ap++;
+```
+
+
+```cairo
+      func main() {
+          %{ memory[ap] = program_input['secret'] %}
+         [ap] = [ap], ap++;
+         ret;
+      }
+``` 
+
+
+
+### WHY HINTS RUN ON SEQUENCERS/ PROVERS AND NOT VERIFIERS
+
+Sequencers are the leads that produces blocks by executing transaction and updating the blockchain state.
+During the proof generation process, the prover uses hints to generate additional constraints or assumptions about the program's behavior, which makes part of the proof sent to the verifier for verification. In the Cairo Virtual Machine, hints are sorted out during different stages of the compilation and execution process by the sequencer. However, they are not directly involved in the verification stage. Hints are primarily for performance optimization.
+    
+The key is the implementation of relevant and accurate hints that assist the prover or sequencers in their respective tasks.
+
+
+
+Example:
+Let's consider a Cairo function that calculates the factorial of a given input *`n`*:
+
+```Cairo
+    @public
+   func factorial(n : felt) -> (result : felt):
+    if n <= 1:
+        result := 1
+    else:
+     result := n * factorial(n - 1)
+```
+
+In this code, the hint *@public* indicates that the *`n`* factorial should be publicly available in the Cairo program.
+The provers/sequencers will utilize the hint to generate a proof or sequence of instructions for the function to ensure maximum optimization. The verifier will check that the generated proof or sequence of instructions has correctly calculated the factorial of the input *`n`* or any other specification is satisfied.
```

Comment:
This is very False. Please, I am hesitant to continue reviewing the PR if you are really not invested in it. Knowing this is false just requires reading the Cairo Docs. Also, writing this simple Cairo Program you'll notice this is untrue.

This kind of content seems AI hallucination, the least you could do is double check if what is being generated is true. Wasting the reviewers time is not a good practice.